akka-cluster-metrics/src/main/scala/akka/cluster/metrics/Provision.scala

```diff
@@ -0,0 +1,317 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.cluster.metrics
+
+import java.io.File
+import kamon.sigar.SigarProvisioner
+import org.hyperic.sigar.Sigar
+import org.hyperic.sigar.SigarProxy
+import org.hyperic.sigar.SigarException
+import scala.language.postfixOps
+
+/**
+ * Provide sigar instance as [[SigarProxy]].
+ *
+ * User can provision sigar classes and native library in one of the following ways:
+ *
+ * 1) Use [[https://github.com/kamon-io/sigar-loader Kamon sigar-loader]] as a project dependency for the user project.
+ * Metrics extension will extract and load sigar library on demand with help of Kamon sigar provisioner.
+ *
+ * 2) Use [[https://github.com/kamon-io/sigar-loader Kamon sigar-loader]] as java agent: `java -javaagent:/path/to/sigar-loader.jar`
+ * Kamon sigar loader agent will extract and load sigar library during JVM start.
+ *
+ * 3) Place `sigar.jar` on the `classpath` and sigar native library for the o/s on the `java.library.path`
+ * User is required to manage both project dependency and library deployment manually.
+ */
+trait SigarProvider {
+
+  /** Library extract location. */
+  def extractFolder: String
+
+  /** Verify if sigar native library is loaded and operational. */
+  def isNativeLoaded: Boolean =
+    try {
+      val sigar = verifiedSigarInstance
+      SigarProvider.close(sigar)
+      true
+    } catch {
+      case e: Throwable ⇒ false
+    }
+
+  /** Create sigar and verify it works. */
+  def verifiedSigarInstance: SigarProxy = {
+    val sigar = new Sigar()
+    sigar.getPid
+    sigar.getLoadAverage
+    sigar.getCpuPerc
+    sigar
+  }
+
+  /** Extract and load sigar native library. */
+  def provisionSigarLibrary(): Unit = {
+    SigarProvisioner.provision(new File(extractFolder))
+  }
+
+  /**
+   *  Create sigar instance with 2-phase sigar library loading.
+   *  1) Assume that library is already provisioned.
+   *  2) Attempt to provision library via sigar-loader.
+   */
+  def createSigarInstance: SigarProxy = {
+    import ProvideScala.Try
+    Try {
+      verifiedSigarInstance
+    } orElse Try {
+      provisionSigarLibrary()
+      verifiedSigarInstance
+    } recover {
+      case e: Throwable ⇒ throw new RuntimeException("Failed to load sigar:", e)
+    } get
+  }
+
+}
+
+object SigarProvider {
+  /**
+   * Release underlying sigar proxy resources.
+   *
+   * Note: [[SigarProxy]] is not [[Sigar]] during tests.
+   */
+  def close(sigar: SigarProxy) = {
+    if (sigar.isInstanceOf[Sigar]) sigar.asInstanceOf[Sigar].close()
+  }
+}
+
+/**
+ * Provide sigar instance as [[SigarProxy]] with configured location via [[ClusterMetricsSettings]].
+ */
+case class DefaultSigarProvider(settings: ClusterMetricsSettings) extends SigarProvider {
+  def extractFolder = settings.NativeLibraryExtractFolder
+}
+
+/**
+ * A copy of [[scala.util.Try]] with the [[scala.util.control.NonFatal]] logic removed.
```

Comment:
well.

1) this is 100-th attempt; and you did not like all the previous ones "with ordinary try-catch"

2) I still hope  [[scala.util.Try]] will get changed, so this class can be removed

3) it makes for nice idiomatic 3 liner everyone will understand
```
    val collector = if (useCustom)
      create(collectorCustom)
    else if (useInternal)
      create(collectorSigar) orElse create(collectorJMX)
    else // Use complete fall back chain.
      create(collectorCustom) orElse create(collectorSigar) orElse create(collectorJMX)
```